### DISCUSSION TITLE
Linking with static library

### DISCUSSION THREAD

**Discussion by rprata** [edubart/nelua-lang#40] [Q&A] [answered]
Hi, I've created my own static library. Is it possible to call lib functions in `nelua`? How can I pass `link-flags` and `include-flags` in script?

**Comment by edubart** [comment, marked as answer]
It is possible, using a dynamic or static library shoud be the same, you need to import its C functions, see how to do that in the C interoperability section in the overview.

As an example, suppose you have `mylib.a` and `mylib.h` in the current folder, that declares the function `mylib_func`:

```lua
## cinclude 'mylib.h'
## linklib 'mylib'
## cflags '-I. -L.' -- tells the C compiler to look for include and library files in the current folder 
local function mylib_func() <cimport, nodecl> end
mylib_func() -- call the library function
```

Also related issues with similar question are #32 and #30.

**Comment by rprata** [comment]
Thank you.
